gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2010,8 +2007,30 @@ private Bucket getBucket(String bucketName) throws IOException {
       if (errorExtractor.itemNotFound(e)) {
         logger.atFiner().withCause(e).log("getBucket(%s): not found", bucketName);
         return null;
+      } else {
+        if (gcsStatisticsMap.get(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+            != null) {
+          gcsStatisticsMap.put(
+              GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES,
```

Comment:
We need to plug in in low-level HTTP client library to count HTTP requests and their responses.

Reply:
Updated